src/sinks/aws_kinesis_firehose.rs

```diff
@@ -170,36 +178,33 @@ impl RetryLogic for KinesisFirehoseRetryLogic {
 enum HealthcheckError {
     #[snafu(display("DescribeDeliveryStream failed: {}", source))]
     DescribeDeliveryStreamFailed {
-        source: RusotoError<rusoto_firehose::DescribeDeliveryStreamError>,
+        source: RusotoError<DescribeDeliveryStreamError>,
     },
     #[snafu(display("Stream name does not match, got {}, expected {}", name, stream_name))]
     StreamNamesMismatch { name: String, stream_name: String },
 }
 
-fn healthcheck(
-    config: KinesisFirehoseSinkConfig,
-    resolver: Resolver,
-) -> crate::Result<super::Healthcheck> {
+async fn healthcheck(config: KinesisFirehoseSinkConfig, resolver: Resolver) -> crate::Result<()> {
     let client = create_client(config.region.try_into()?, config.assume_role, resolver)?;
```

Comment:
I think we want the `region.try_into()?` to happen outside of the `async fn` so as to catch config errors up front rather than having them happen after the system starts running.

Reply:
`region.try_into()?` also happen on creating `KinesisFirehoseService` which is not async, so config error should be catched up front.
https://github.com/timberio/vector/pull/2735/files#diff-6b010670b04b03f3921950a0b304868dR76